src/components/data/DataTable.tsx

```diff
@@ -75,6 +100,38 @@ function defaultRenderItem<Row>(r: Row, field: keyof Row): ComponentChildren {
   return r[field] as ComponentChildren;
 }
 
+function calculateNewOrder<T>(newField: T, prevOrder: Order<T>): Order<T> {
+  if (newField !== prevOrder?.field) {
+    return { field: newField, direction: 'ascending' };
+  }
+
+  return prevOrder.direction === 'ascending'
+    ? { field: newField, direction: 'descending' }
+    : null;
+}
+
+type HeaderComponentProps = {
+  onClick?: () => void;
+  children: ComponentChildren;
+  label: string;
+};
+
+function HeaderComponent({ children, onClick, label }: HeaderComponentProps) {
+  const commonClasses = 'flex justify-between items-center';
+  return onClick ? (
+    <Button
+      classes={`${commonClasses} w-full !p-3`}
+      variant="custom"
+      onClick={onClick}
+      title={`Click to order table by ${label}`}
```

Comment:
I think we can omit this title here, as it is a strong convention that tables with button headers will sort their columns. Including the "Click to order table by" text makes navigating the table a bit annoying because it adds uninformative text in front of the column name. The example you mentioned at https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/ for example just uses regular buttons. There is also a single table-wide caption to provide a hint, but I don't think that's essential.

Reply:
I had mixed feelings about this title, but I also didn't love the caption approach suggested in https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/, so if you think we can just skip this, I'm happy with it 🙂

Reply:
I think this is a common-enough pattern that we can skip the caption.